src/dbnode/storage/entry.go

```diff
@@ -173,6 +178,38 @@ func (entry *Entry) ReconciledOnIndexSeries() (doc.OnIndexSeries, resource.Simpl
 	}), true
 }
 
+// GetEntryIndexBlockStates returns the entry index block states by time for
+// the indexed entry.
+func (entry *Entry) GetEntryIndexBlockStates() doc.EntryIndexBlockStates {
```

Comment:
Is there a reason for this `Get`? IMO we can just have a single merge func that takes in the other entry.

Reply:
Also I think we can avoid this map copy alloc and just mutate the entry's map directly that we call merge on

Reply:
Yeah removed this after the refactor to move to the segments builder:+1: